src/containers/Folder.jsx

```diff
@@ -67,35 +69,24 @@ class Folder extends Component {
 
 const mapStateToProps = (state, ownProps) => ({
   isFetching: state.ui.isFetching,
+  showLoading: state.ui.isFetching && state.folder === null,
   alert: state.ui.alert,
   error: state.ui.error,
   showSelection: mustShowSelectionBar(state),
   showDeleteConfirmation: state.ui.showDeleteConfirmation,
   showActionMenu: state.ui.showFileActionMenu,
-  files: getVisibleFiles(state),
-  folderId: state.ui.currentFolderId
+  files: getVisibleFiles(state)
 })
 
 const mapDispatchToProps = (dispatch, ownProps) => ({
   onMount: () => {
-    let folderId
-    if (ownProps.params.file !== undefined) {
-      folderId = ownProps.params.file
-    } else {
-      folderId = ownProps.context === TRASH_CONTEXT
-        ? TRASH_DIR_ID
-        : ROOT_DIR_ID
-    }
-    dispatch(openFolder(folderId, true))
+    dispatch(openFolder(ownProps.params.folderId, ownProps.context))
   },
-  onRouteChange: (folderId) => {
-    dispatch(openFolder(folderId, true))
+  onRouteChange: (folderId, context) => {
+    dispatch(openFolder(folderId, context))
   },
   onFolderOpen: (folderId) => {
-    return dispatch(openFolder(folderId, false))
-      .then(() => {
-        ownProps.router.push(`/${ownProps.context}/${folderId}`)
-      })
+    return dispatch(openFolder(folderId, ownProps.context))
   },
```

Comment:
And same here, and in addition arrow function returns the first instruction's result:

```javascript
onFolderOpen: folderId => dispatch(openFolder(folderId, ownProps.context))
```